File: 프로그래머스/0/120907. OX퀴즈/OX퀴즈.js
function solution(quiz) {
    const answer = [];
    
    for(let q of quiz) {
        let [first, oper, second, result] = q.replaceAll("= ", "").split(" ");
        
        let compare;
        if(oper==="+") {
            compare = parseInt(first) + parseInt(second);
        }
        if(oper==="-") {
            compare = parseInt(first) - parseInt(second);
        }
        
        answer.push(compare.toString()===result ? "O" : "X");
    }
    
    return answer;
}

File: 프로그래머스/0/120907. OX퀴즈/OX퀴즈.py
def solution(quiz):
    answer = []

    for expression in quiz:
        lst = expression.split()

        if lst[1] == '+':
            if int(lst[0]) + int(lst[2]) == int(lst[4]):
                answer.append("O")
            else:
                answer.append("X")
        elif lst[1] == '-':
            if int(lst[0]) - int(lst[2]) == int(lst[4]):
                answer.append("O")
            else:
                answer.append("X")

    return answer